From 397ef20e3d8354dbd70c022f249101c106fe69ea Mon Sep 17 00:00:00 2001
From: chandavath <rcrajagopal@habana.ai>
Date: Fri, 09 Feb 2024 14:15:41 +0200
Subject: [PATCH] [SW-137122] gaudi3: erom rec: pll config

Faced slowness during firmware update in erom recovery,
while copying fit to sram. So required pll's are
configured to overcome the slowness.

Change-Id: I4497d9ccc43ba444fa6b74453986cf322545fb36
Signed-off-by: chandavath <rcrajagopal@habana.ai>
---

diff --git a/arch/gaudi3/common/asic.c b/arch/gaudi3/common/asic.c
index 64e238b..3fa3dcc 100644
--- a/arch/gaudi3/common/asic.c
+++ b/arch/gaudi3/common/asic.c
@@ -2,3 +2,66 @@
 /*
  * Copyright (C) 2022 HabanaLabs Ltd.
  */
+
+#include <gaudi3_blocks_linux_driver.h>
+#include <pll.h>
+#include <baremetal.h>
+
+static struct hl_pll_cfg gaudi3_pll_cfg[] = {
+	{
+		.name = "G0MSH",
+		.base = mmD0_G0_MSH_PLL_CTRL_BASE,
+		.nr = 1, .nf = 224, .od = 6, .nb = 0,
+		.div_factor = { 0, 1, 1, 7 },
+		.div_sel = { HL_PLL_PLL_CLK,
+				 HL_PLL_DIV_PLL,
+				 HL_PLL_DIV_PLL,
+				 HL_PLL_DIV_PLL
+		},
+	}, {
+		.name = "G6MSH",
+		.base = mmD0_G6_MSH_PLL_CTRL_BASE,
+		.nr = 1, .nf = 224, .od = 6, .nb = 0,
+		.div_factor = { 0, 1, 1, 7 },
+		.div_sel = { HL_PLL_PLL_CLK,
+				 HL_PLL_DIV_PLL,
+				 HL_PLL_DIV_PLL,
+				 HL_PLL_DIV_PLL
+		},
+	}, {
+		.name = "G4NCH",
+		.base = mmD0_G4_NCH_PLL_CTRL_BASE,
+		.nr = 1, .nf = 224, .od = 6, .nb = 0,
+		.div_factor = { 0, 1, 1, 7 },
+		.div_sel = { HL_PLL_PLL_CLK,
+				 HL_PLL_DIV_PLL,
+				 HL_PLL_DIV_PLL,
+				 HL_PLL_DIV_PLL
+		},
+	}, {
+		.name = "G1CS",
+		.base = mmD0_G1_CS_PLL_CTRL_BASE,
+		.nr = 1, .nf = 240, .od = 3, .nb = 2,
+		.div_factor = { 0, 1, 2, 7 },
+		.div_sel = { HL_PLL_PLL_CLK,
+				 HL_PLL_DIV_PLL,
+				 HL_PLL_DIV_PLL,
+				 HL_PLL_DIV_PLL
+		},
+	}, {
+		.name = "G5CS",
+		.base = mmD0_G5_CS_PLL_CTRL_BASE,
+		.nr = 1, .nf = 240, .od = 3, .nb = 2,
+		.div_factor = { 0, 1, 2, 7 },
+		.div_sel = { HL_PLL_PLL_CLK,
+				 HL_PLL_DIV_PLL,
+				 HL_PLL_DIV_PLL,
+				 HL_PLL_DIV_PLL
+		},
+	}
+};
+
+void hl_dev_pll_init(void)
+{
+	hl_pll_init(gaudi3_pll_cfg, ARRAY_SIZE(gaudi3_pll_cfg));
+}
